### PR TITLE
Add integration test for user store rest API

### PR DESCRIPTION
**Pull request by Kanapriya**
Refer - https://github.com/wso2/product-is/issues/6830

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SUS-50012");
+    }
+
+    @Test
+    public void testUpdateUserStoreByIdNegativeCase() throws IOException {
+
+        String body = readResource("add-secondary-user-store-negative.json");
+        Response response = getResponseOfPut(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR +
+                INCORRECT_DOMAIN_ID, body);
+        validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, "SUS-50004");
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: This should not have returned internal server error. If the domain is incorrect, it should return not found. Fix the endpoint side to do so.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        response.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK)
+                .body("", Matchers.hasSize(0));
+    }
+
+    @Test
+    public void testDeleteUserStoreByIdNegativeCase() {
+
+        Response response = getResponseOfDelete(USER_STORE_PATH_COMPONENT + PATH_SEPARATOR
+                + "SkRCQy0DRETYUI");
+        validateErrorResponse(response, HttpStatus.SC_INTERNAL_SERVER_ERROR, "SUS-50002");
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: This should not have returned internal server error. If the domain is incorrect, it should return not found. Fix the endpoint side to do so.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/model/AvailableUserStoreClassesRes.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                Objects.equals(this.className, availableUserStoreClassesRes.className);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(typeId, typeName, className);
+    }
+
+    @Override
+    public String toString() {
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/model/AvailableUserStoreClassesRes.java]: `toString` and `toIndentedString` are not required for the test.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreSuccessTest.java [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+            validatableResponse.body("find{ it.id == '" + resEntry.getKey() + "' }.self",
+                    equalTo("/t/" + tenant + "/api/server/v1" + USER_STORE_PATH_COMPONENT +
+                            PATH_SEPARATOR + resEntry.getValue().getId()));
+        }
+    }
+
+    @Test(dependsOnMethods = {"testGetUserStore"})
+    public void testGetUserStoreTypeMeta() {
+
+        String typeId = "SkRCQ1VzZXJTdG9yZU1hbmFnZXI";
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreSuccessTest.java]: Constant

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java (hunk 2 above)
Comment (Kanapriya) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: Fixed in a50eaccbf9929d07a1c6542b920d4add8bed472a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java (hunk 1 above)
Comment (Kanapriya) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: Fixed in a50eaccbf9929d07a1c6542b920d4add8bed472a

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                INCORRECT_DOMAIN_ID, body);
+        validateErrorResponse(response, HttpStatus.SC_NOT_FOUND, "SUS-60001");
+    }
+
+    @Test
+    public void testGetUserStoreTypeByIdNegativeCase() {
+
+        Response response = getResponseOfGet(USER_STORE_PATH_COMPONENT + USER_STORE_META_COMPONENT
+                + PATH_SEPARATOR + INCORRECT_TYPE_ID);
+        response.then().log().ifValidationFails().assertThat().statusCode(HttpStatus.SC_OK)
```
Comment (madurangasiriwardena) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: if the id is not correct, it should return not found

Comment (Kanapriya) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/user/store/v1/UserStoreFailureTest.java]: Fix with b68f8e3b41be55fe2c21fca74b10ce550402038a

**Pull request merged**